### PR TITLE
feat: add getItemsByFunction and getItemsByFunctionAtSlot

### PR DESCRIPTION
**Pull request by LordMidas**
To be merged AFTER #316 (port to modern hooks)

### CODE REVIEW

File: msu/hooks/items/item_container.nut [hunk 1]
```diff
@@ -94,4 +94,35 @@
...
 		return unequip(_item);
 	}
+
+	q.getItemsByFunction <- function( _function )
+	{
+		local ret = [];
+
+		foreach (slot, spaces in ::Const.ItemSlotSpaces)
+		{
+			for (local i = 0; i < spaces; i++)
```
Comment (TaroEld) [inline msu/hooks/items/item_container.nut:104]: This should probably just use `getItemsByFunctionAtSlot`:
`ret.extend(getItemsByFunctionAtSlot(slot, _function);`

Comment (LordMidas) [reply msu/hooks/items/item_container.nut:104]: That results in way more iterations and array declarations than the current implementation. I don't like that.

Comment (TaroEld) [reply msu/hooks/items/item_container.nut:104]: Is it worse than resusing the same code in two functions with the only difference being another loop though

Comment (TaroEld) [reply msu/hooks/items/item_container.nut:104]: nvm

File: msu/hooks/items/item_container.nut [hunk 2]
```diff
@@ -92,4 +92,36 @@
...
+			{
+				this.__putItemsAtSlotIntoArray(slot, ret);
+			}
```
Comment (TaroEld) [inline msu/hooks/items/item_container.nut:105]: Wait, I don't get it. Why this inner loop? Shouldn't this just run for all the slots, and the private function then iterates over the items in that slot?

Comment (LordMidas) [reply msu/hooks/items/item_container.nut]: Fixed

**Review by TaroEld — APPROVED** [review]

**Review by TaroEld — APPROVED** [review]

**Pull request merged**